api/buffer/design.go

```diff
@@ -0,0 +1,549 @@
+	})
+	Files("/documentation/openapi3.yaml", "openapi3.yaml", func() {
+		Meta("openapi:summary", "OpenAPI 3.0 YAML Specification")
+		Meta("openapi:tag:documentation")
+	})
+	Files("/documentation/{*path}", "swagger-ui", func() {
+		Meta("openapi:generate", "false")
+		Meta("openapi:summary", "Swagger UI")
+		Meta("openapi:tag:documentation")
+	})
+
+	// Main endpoints ---------------------------------------------------------------------------------------------
+
+	Method("CreateReceiver", func() {
+		Meta("openapi:summary", "Create receiver")
+		Description("Create a new receiver for a given project")
+		Result(Receiver)
+		Payload(func() {
+			receiverId()
+			name("receiver")
+			Attribute("exports", ArrayOf(Export), "List of receiver exports. A receiver may have a maximum of 20 exports.")
+			Required("name", "exports")
+		})
+		HTTP(func() {
+			POST("/receivers")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+		})
+	})
+
+	Method("ListReceivers", func() {
+		Meta("openapi:summary", "List all receivers")
+		Description("List all receivers for a given project.")
+		Result(ArrayOf(Receiver))
+		HTTP(func() {
+			GET("/receivers")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+		})
+	})
+
+	Method("GetReceiver", func() {
+		Meta("openapi:summary", "Get receiver")
+		Description("Get the configuration of a receiver.")
+		Result(Receiver)
+		Payload(func() {
+			receiverId()
+			Required("receiverId")
+		})
+		HTTP(func() {
+			GET("/receivers/{receiverId}")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+			ReceiverNotFoundError()
+		})
+	})
+
+	Method("DeleteReceiver", func() {
+		Meta("openapi:summary", "Delete receiver")
+		Description("Delete a receiver.")
+		Result(Receiver)
+		Payload(func() {
+			receiverId()
+			Required("receiverId")
+		})
+		HTTP(func() {
+			DELETE("/receivers/{receiverId}")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+			ReceiverNotFoundError()
+		})
+	})
+
+	Method("RefreshReceiverTokens", func() {
+		Meta("openapi:summary", "Refresh receiver tokens")
+		Description("Refresh all tokens used by a receiver.")
+		Result(Receiver)
+		Payload(func() {
+			receiverId()
+			Required("receiverId")
+		})
+		HTTP(func() {
+			POST("/receivers/{receiverId}/tokens/refresh")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+			ReceiverNotFoundError()
+		})
+	})
+
+	Method("CreateExport", func() {
+		Meta("openapi:summary", "Create export")
+		Description("Create a new export for an existing receiver.")
+		Result(Receiver)
+		Payload(func() {
+			receiverId()
+			Attribute("export", Export)
+			Required("receiverId", "export")
+		})
+		HTTP(func() {
+			POST("/receivers/{receiverId}/exports")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+			ReceiverNotFoundError()
+		})
+	})
+
+	Method("UpdateExport", func() {
+		Meta("openapi:summary", "Update export")
+		Description("Update a receiver export. The export's columns must match the existing table schema.")
```

Comment:
`The export's columns must match the existing table schema.`
Nemusia, moze sa zmenit cielova tabulka spolu s `columns`.
Ulozi sa nova revizia mappingu, na ktoru sa vsetky API nodes switchnu a budu zapisovat tam.
Tj. staci tu vetu zmazat.

Reply:
Našel jsem to tady v design dokumentu:
![image](https://user-images.githubusercontent.com/1665677/199506923-a8cf97cb-30cd-495e-b578-7b6825c0402c.png)
Mám to teda vymazat i tam?

Reply:
jj, zmaz to, ako sa to postupne doplnovalo, tak som si nevsimol, ze to je potrebne zmazat aj tu.